core/javatests/com/google/instrumentation/stats/ViewTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import static org.junit.Assert.assertTrue;
+
+import com.google.instrumentation.common.Duration;
+import com.google.instrumentation.common.Function;
+import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.stats.DistributionAggregation.Range;
+import com.google.instrumentation.stats.IntervalAggregation.Interval;
+import com.google.instrumentation.stats.MeasurementDescriptor.BasicUnit;
+import com.google.instrumentation.stats.MeasurementDescriptor.MeasurementUnit;
+import com.google.instrumentation.stats.View.DistributionView;
+import com.google.instrumentation.stats.View.IntervalView;
+import com.google.instrumentation.stats.ViewDescriptor.DistributionViewDescriptor;
+import com.google.instrumentation.stats.ViewDescriptor.IntervalViewDescriptor;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for class {@link View}.
+ */
+@RunWith(JUnit4.class)
+public final class ViewTest {
+  @Test
+  public void testDistributionView() {
+    DistributionAggregationDescriptor aggregationDescriptor =
+        DistributionAggregationDescriptor.create();
+    final DistributionViewDescriptor viewDescriptor =
+        DistributionViewDescriptor.create(
+            name, description, measurementDescriptor, aggregationDescriptor, tagKeys);
+    List<DistributionAggregation> aggregations = Arrays.asList(new DistributionAggregation[] {
+          DistributionAggregation.create(tags1, 5, 5.0, 15.0, Range.create(1.0, 5.0),
+              Arrays.asList(new Long[] { 1L, 1L, 1L, 1L, 1L })),
+          DistributionAggregation.create(tags2, 10, 5.0, 30.0, Range.create(1.0, 5.0),
+              Arrays.asList(new Long[] { 2L, 2L, 2L, 2L, 2L }))
+        });
+    final Timestamp start = Timestamp.fromMillis(1000);
+    final Timestamp end = Timestamp.fromMillis(2000);
+    final View view = DistributionView.create(viewDescriptor, aggregations, start, end);
```

Comment:
Yes, N boundaries defines N+1 buckets. Technically, this is fine because the final bucket is implicitly 0, but I've fixed it to be clear.